src/vmemcache.c

```diff
@@ -69,6 +173,61 @@ vmemcache_newU(const char *dir, size_t max_size, size_t fragment_size,
 void
 vmemcache_delete(VMEMcache *cache)
 {
+	while (vmemcache_evict(cache, NULL, 0) == 0)
+		;
+	repl_p_destroy(&cache->repl);
+	ravl_delete(cache->index);
+	vmcache_heap_destroy(cache->heap);
+	munmap(cache->addr, cache->size);
+	Free(cache);
+}
+
+/*
+ * vmemcache_populate_fragments -- (internal) populate fragments
+ */
+static void
+vmemcache_populate_fragments(struct cache_entry *entry,
+				const char *value, size_t value_size)
+{
+	struct heap_entry he;
+	size_t size_left = value_size;
+
+	VEC_FOREACH(he, &entry->value.fragments) {
+		if (he.size <= size_left) {
+			memcpy(he.ptr, value, he.size);
+			value += he.size;
+			size_left -= he.size;
+			if (size_left == 0)
```

Comment:
We shouldn't need either of these breaks - the loop will terminate because this MUST be the last fragment. Add an equivalent ASSERT.

Reply:
Done